src/snapred/backend/error/StateValidationException.py

```diff
@@ -11,9 +11,14 @@
 class StateValidationException(Exception):
     "Raised when an Instrument State is invalid"
 
-    def __init__(self, exception: Exception):
-        exceptionStr = str(exception)
-        tb = exception.__traceback__
+    def __init__(self, exception):
+        # Handle both string and Exception types for 'exception'
+        if isinstance(exception, Exception):
```

Comment:
I'm not sure we should be expanding exceptions type to be both `Exception` and `str`, I dont think I see the usecase in this pr?  I just see this used in a test file.

Reply:
It is used in the test.

Reply:
This has been reverted